lib/src/git_backend.rs

```diff
@@ -406,7 +412,27 @@ fn commit_from_git_without_root_parent(
     let author = signature_from_git(commit.author());
     let committer = signature_from_git(commit.committer());
 
-    Commit {
+    // If the commit is signed, extract both the signature and the signed data
+    // (which is the commit buffer with the gpgsig header omitted).
+    // We have to re-parse the raw commit data because gix CommitRef does not give
+    // us the sogned data, only the signature.
+    // Ideally, we could use try_to_commit_ref_iter at the beginning of this
```

Comment:
Perhaps, we can use (owned, but not parsed) `gix::Commit` object instead?
https://docs.rs/gix/0.55.2/gix/struct.Commit.html#method.signature

I used `.try_to_commit_ref()` instead of `.try_into_commit()` (and `.decode()`) just because it was easier and good enough for the existing code.

Reply:
oh, I did not find this

Sure this will be good, let me try this

Reply:
Huh, `git::Commit` actually parses the commit object from scratch on every call.

So the comment's still true, the optimal thing to do is to use CommitRefIter and get all the data we need in a single pass.

For now I guess the current version is fine?.
Anything signing related is very WIP.
Guess I should add a todo for it to be actionable at some point.

Reply:
Ah, yes. Peeking `.extra_headers` should be cheaper than `.decode()` + `.signature()`.

> For now I guess the current version is fine?

+1

Reply:
Actually, CommitRefIter does not yield token ranges, which we'd need to cut out the gpgsig to get the signed data.
So the only way to get both the signature and the signed data from gix currently is `CommitRefIter::signature`, which we're using, which does re-parse commit data 🤷
(but only when gix already have found that there _is_ a signature in the first pass, at least)